runtime/oti/cfr.h

```diff
@@ -890,7 +891,7 @@ typedef struct J9CfrClassFile {
 #define CFR_MINOR_VERSION  3
 #define CFR_PUBLIC_PRIVATE_PROTECTED_MASK	(CFR_ACC_PUBLIC | CFR_ACC_PRIVATE | CFR_ACC_PROTECTED)
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_SUPER | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM | CFR_ACC_VALUE_TYPE)
+#define CFR_CLASS_ACCESS_MASK					(CFR_ACC_PUBLIC | CFR_ACC_FINAL | CFR_ACC_SUPER | CFR_ACC_INTERFACE | CFR_ACC_ABSTRACT | CFR_ACC_SYNTHETIC | CFR_ACC_ANNOTATION | CFR_ACC_ENUM | CFR_ACC_VALUE_TYPE | CFR_ACC_ATOMIC)
```

Comment:
Is this going to cause problems in older JDK levels (i.e. checking for bad modifiers being set)?

Reply:
I've looked at where `CFR_CLASS_ACCESS_MASK` is used. I don't see it causing such problems. However, I have been testing this change only on Java 17 so far. I can launch a Java 8 build. Will update here once my Java 8 build finishes.

Reply:
The problem is that the 0x40 bit is currently not allowed in class modifiers. Adding the bit unconditionally to the mask will allow the invalidly-set bit through in older JDKs.

Sadly, this will need to be a runtime classfile version check, not an ifdef on JCL version. Probably the easiest change is to leave this constant alone and augment it with the new bit at the check site if the classfile version is high enough.

Reply:
OK. I see what you mean here. I think the same problem also applies to `CFR_ACC_VALUE_TYPE`. I can add a class file version check in `cfreader.c` before accepting `CFR_ACC_ATOMIC`.

Reply:
> I can add a class file version check in cfreader.c before accepting CFR_ACC_ATOMIC, CFR_ACC_VALUE_TYPE.

Done. 

If such flags are found in an older version class file, I simply ignore them rather than throwing an exception, because the spec says:

```
In a class file with any other version number, the 0x0100 bit is ignored, and the ACC_PRIMITIVE flag is considered not to be set.
In a class file with any other version number, the 0x0040 bit is ignored, and the ACC_ATOMIC flag is considered not to be set.
```

Reply:
That looks like a recent specification - I'm concerned about older versions.

Reply:
Yes, it is from a recent spec of Java 17. 

Looking at Java 11 JVM sepc: https://docs.oracle.com/javase/specs/jvms/se11/html/jvms-4.html#jvms-4.1-200-E.1, it says:

`All bits of the access_flags item not assigned in Table 4.1-B are reserved for future use. They should be set to zero in generated class files and should be ignored by Java Virtual Machine implementations. `

Java 8 JVM spec also have the same message.

Reply:
OK